### PR TITLE
Fixes #97 and #70 - fix gcp db connection and implement flyway

### LINKED ISSUE #97: Fix GCP DB connection
app won't start in cloud run due to some misconfiguration

### PR DESCRIPTION
**Pull request by mkimberlin**
Or at least it has a good chance to...so let's find out! :-)

### CODE REVIEW

**Comment by mkimberlin** [comment]
Odd...not sure how that build error got in if it didn't happen on Jesse's PR. I'll look to fix...didn't touch that part of the app though. :-\

**Comment by jorwalk** [comment]
```
Line 116:15:  React Hook useMemo does nothing when called with only one argument. Did you forget to pass an array of dependencies?  react-hooks/exhaustive-deps
```

File: server/src/main/resources/application.yml [hunk 1]
```diff
@@ -64,4 +47,22 @@ micronaut:
       login-success-target-url: '/'
       unauthorized-target-url: '/unauthorized'
       forbidden-target-url: '/forbidden'
-      
\ No newline at end of file
+---      
+datasources:
+  default:
+    driver-class-name: org.postgresql.Driver
+    dialect: POSTGRES
```
Comment (bhatiam12) [inline server/src/main/resources/application.yml:54]: we don't need to pass username and password ?

File: server/src/main/resources/application-local.yml [hunk 2]
```diff
@@ -1,66 +1,5 @@
...
-
-  server:
-    cors:
-      enabled: true
-    maxRequestSize: 100MB
-    multipart:
-      maxFileSize: 100MB
-
-  router:
-    static-resources:
```
Comment (bhatiam12) [inline server/src/main/resources/application-local.yml:13]: why did we delete all the config from application local ? is it for testing ?

File: server/src/main/resources/application.yml (hunk 1 above)
Comment (mkimberlin) [reply server/src/main/resources/application.yml:54]: It should be picked up from the ENV, I believe. I may have to change the env var passed to DATASOURCES_DEFAULT_PASSWORD and DATASOURCES_DEFAULT_USERNAME, but I'll see on deploy.

File: server/src/main/resources/application-local.yml (hunk 2 above)
Comment (mkimberlin) [reply server/src/main/resources/application-local.yml:13]: These files are cascading. You don't need to redefine all of the properties...only the ones that need to change. Check out the micronaut configuration documentation, it talks about order of precedence for properties.

File: server/src/main/resources/application.yml (hunk 1 above)
Comment (mkimberlin) [reply server/src/main/resources/application.yml:54]: Just altered this...pretty positive it was going to cause an error.

File: server/src/main/resources/application-local.yml (hunk 2 above)
Comment (bhatiam12) [reply server/src/main/resources/application-local.yml:13]: No, I meant swagger properties from local yaml. I don't see them defined anywhere.

Comment (bhatiam12) [reply server/src/main/resources/application-local.yml:13]: sorry, should have viewed the complete application.yaml and not just the changes :P

File: server/src/main/java/com/objectcomputing/checkins/gcp/postgres/GoogleCloudDatabaseSetup.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import io.micronaut.configuration.jdbc.hikari.DatasourceConfiguration;
+import io.micronaut.context.annotation.Property;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.env.Environment;
+import io.micronaut.context.event.BeanCreatedEvent;
+import io.micronaut.context.event.BeanCreatedEventListener;
+
+@Singleton
+@Requires(env = Environment.GOOGLE_COMPUTE)
```
Comment (mkimberlin) [inline server/src/main/java/com/objectcomputing/checkins/gcp/postgres/GoogleCloudDatabaseSetup.java:13]: @Maroju100 Here's your answer to limiting the basic auth to tests: Environment.TEST

**Pull request merged**